torch/csrc/jit/tracer.cpp

```diff
@@ -227,14 +227,28 @@ std::pair<std::shared_ptr<TracingState>, Stack> enter(Stack inputs) {
         elems[i] = add_input(elems[i], elem_types[i], elem_values[i]);
       }
       return Tuple::create(std::move(elems));
+    } else if (auto dict_type = type->cast<DictType>()) {
+      auto elem_pairs = input.toGenericDict()->elements();
+      auto unpack_to_list = state->graph->insert(aten::values, {value});
```

Comment:
Not sure I fully understand - what does inserting the aten::values & list unpack achieve exactly ?

Reply:
If it makes no sense here, it also makes no sense for tuples. I admit that I'm mostly cargo-culting, but my understanding was that we flatten everything and then pack it back up so we can keep track of all the tensors involved.